framework/tlvf/AutoGenerated/src/tlvf/ieee_1905_1/tlvWscM2.cpp

```diff
@@ -454,7 +454,10 @@ bool tlvWscM2::add_encrypted_settings(std::shared_ptr<WSC::cWscAttrEncryptedSett
         TLVF_LOG(ERROR) << "No call to create_encrypted_settings was called before add_encrypted_settings";
         return false;
     }
-    if (ptr->getStartBuffPtr() != (uint8_t *)m_encrypted_settings) {
+    uint8_t *src = (uint8_t*) m_encrypted_settings;
```

Comment:
This doesn't look right...

Reply:
why not?

Reply:
because `m_encrypted_settings` is a pointer to the start of the list, but if this is not the first `create`-`add` then the src pointer should point to the last entry. I fixed this in #140

Reply:
`src` points, at first, to `m_encrypted_settings`, that's the default value (in case that's the first create-add).
After that, the code checks if there are already created entries, and if so changes the value of `m_encrypted_settings`.
You don't see this code here because `m_encrypted_settings` is not a list but a single class, so there's no point to check the length of the list.
Look at line 245 in [tlvVarList.cpp](https://github.com/prplfoundation/prplMesh/pull/139/files/7e1c456e4daef585dea05e61209a637a5c0ce933#diff-63f6521e0ba10e43bad9fbf707ec4087) and you'll understand what I mean.